include/vi.h

```diff
@@ -0,0 +1,240 @@
+/**
+ * @file vi.h
+ * @brief Video Interface Subsystem
+ * @ingroup display
+ */
+#ifndef __LIBDRAGON_VI_H
+#define __LIBDRAGON_VI_H
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "regsinternal.h"
+#include "n64sys.h"
+
+/**
+ * @addtogroup display
+ * @{
+ */
+
+typedef uint32_t vi_register_t;
+
+/** @brief Register uncached location in memory of VI */
+#define VI_REGISTERS_ADDR       0xA4400000
+/** @brief Number of useful 32-bit registers at the register base */
+#define VI_REGISTERS_COUNT      14
+
+/**
+ * @brief Video Interface register structure
+ *
+ * Whenever trying to configure VI registers, 
+ * this struct and its index definitions below can be very useful 
+ * in writing comprehensive and verbose code.
+ */
+typedef struct vi_config_s{
+    vi_register_t regs[VI_REGISTERS_COUNT];
+} vi_config_t;
+
+volatile vi_register_t* VI_REGISTERS    =  ((volatile vi_register_t*)VI_REGISTERS_ADDR);
```

Comment:
This definition of a variable at a global scope in a header file will cause duplicate variable definitions once we start including this file in different source files. I suggest we stick to the standard approach of using macros instead.